kura/examples/org.eclipse.kura.example.ble.tisensortag.tinyb/src/main/java/org/eclipse/kura/example/ble/tisensortag/tinyb/TiSensorTag.java

```diff
@@ -539,21 +552,29 @@ public void disableHumidityNotifications() {
      */
     public void setHygrometerPeriod(int period) {
         byte[] periodBytes = { ByteBuffer.allocate(4).putInt(period).array()[3] };
-        BluetoothLeGattCharacteristic humPeriodChar;
         try {
-            humPeriodChar = this.gattServices.get(HUMIDITY).findCharacteristic(TiSensorTagGatt.UUID_HUM_SENSOR_PERIOD);
-            humPeriodChar.writeValue(periodBytes);
+            this.gattResources.get(HUMIDITY).getGattService().findCharacteristic(TiSensorTagGatt.UUID_HUM_SENSOR_PERIOD)
+                    .writeValue(periodBytes);
         } catch (KuraException e) {
             logger.error("Hygrometer period set failed", e);
         }
     }
 
+    public boolean isHygrometerNotifying() {
+        TiSensorTagGattResources resource = this.gattResources.get(HUMIDITY);
+        if (resource != null) {
```

Comment:
maybe this can be externalised to a method since it's the same code for all the types of sensors